receiver/sqlserverreceiver/queries.go

```diff
@@ -351,3 +351,632 @@ var sqlServerQuerySamples string
 func getSQLServerQuerySamplesQuery() string {
 	return sqlServerQuerySamples
 }
+
+// Conditional check based on Azure SQL DB v/s the rest aka (Azure SQL Managed instance OR On-prem SQL Server)
+// EngineEdition=5 is Azure SQL DB
+const sqlServerWaitStatsQuery string = `
+SET DEADLOCK_PRIORITY -10;
+
+IF SERVERPROPERTY('EngineEdition') != 5
+SELECT
+	'sqlserver_waitstats' AS [measurement],
+REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
+DB_NAME() as [database_name],
+ws.wait_type,
+wait_time_ms,
+wait_time_ms - signal_wait_time_ms AS [resource_wait_ms],
+signal_wait_time_ms,
+max_wait_time_ms,
+waiting_tasks_count,
+ISNULL(wc.wait_category,'OTHER') AS [wait_category]
+FROM
+sys.dm_os_wait_stats AS ws WITH (NOLOCK)
+LEFT OUTER JOIN ( VALUES
+('ASYNC_IO_COMPLETION','Other Disk IO'),
```

Comment:
Have we considered moving this conversion to Go code to simplify the SQL? This approach could also make it easier for others to reuse this section if needed.

Reply:
We could potentially have the logic done in Go, but I'd think that would actually make the logic more complicated. Doing a query and then manually modifying results seems like a more involved process than the query by itself. Do you think there would be considerable performance improvements, or do you see another place where the logic could be reused?

This query is also coming from Telegraf, which allows us to be downstream from them if any changes come up. Modifying the query would mean we could no longer stay up to date with Telegraf, and add a potential maintenance burden here.

Unless there's a considerable downside to the performance of the query as it is, I'd prefer to do this logic in the query.

Reply:
I'm fine with this solution. We can consider migrating it to Go if we need to use similar logic somewhere else.